src/ctia/store.clj

```diff
@@ -85,40 +85,21 @@
   (create-identity [this new-identity])
   (delete-identity [this org-id role]))
 
-;; core model
-(defonce judgement-store (atom nil))
-(defonce indicator-store (atom nil))
-(defonce feedback-store (atom nil))
-
-;; threats
-(defonce ttp-store (atom nil))
-(defonce campaign-store (atom nil))
-(defonce actor-store (atom nil))
-(defonce coa-store (atom nil))
-(defonce exploit-target-store (atom nil))
-
-;; sightings
-(defonce sighting-store (atom nil))
-
-;; incidents
-(defonce incident-store (atom nil))
-
-;; relations
-(defonce relation-store (atom nil))
-
-;; internal
-(defonce identity-store (atom nil))
-
-(def stores
-  {:judgement judgement-store
-   :indicator indicator-store
-   :feedback feedback-store
-   :ttp ttp-store
-   :campaign campaign-store
-   :actor actor-store
-   :coa coa-store
-   :exploit-target exploit-target-store
-   :sighting sighting-store
-   :incident incident-store
-   ;;:relation relation-store
-   :identity identity-store})
+(defonce stores (atom {:judgement []
+                       :indicator []
+                       :feedback []
+                       :ttp []
+                       :campaign []
+                       :actor []
+                       :coa []
+                       :exploit-target []
+                       :sighting []
+                       :incident []
+                       ;;:relation relation-store
+                       :identity []}))
+
+(defn write-store [e write-fn]
+  (first (doall (map write-fn (e @stores)))))
+
+(defn read-store [e read-fn]
+  (read-fn (rand-nth (e @stores))))
```

Comment:
I would use:

~~~clojure
(defn read-store [store write-fn & args]
    (first (doall (map (apply write-fn (e @stores) args))))))
~~~

Which could replace

~~~clojure
(read-store :sighting (fn [s] (read-sighting s id)))
~~~

by

~~~clojure
(read-store :sighting read-sighting id)
~~~

The same remarks can be done for the `write-store` function.

Also as the `read-store` and `write-store` read and write to many store, I would have put a plural like: `read-stores`, `write-stores` or `read-from-stores`, `write-to-stores`.